nym-wallet/src-tauri/src/wallet_storage/mod.rs

```diff
@@ -170,12 +192,27 @@ fn store_login_with_multiple_accounts_at_file(
         return Err(BackendError::WalletDifferentPasswordDetected);
     }
 
-    let mut new_accounts = MultipleAccounts::new();
-    new_accounts.add(DEFAULT_FIRST_ACCOUNT_NAME.into(), mnemonic, hd_path)?;
-    let new_login = StoredLogin::Multiple(new_accounts);
-    let new_encrypted_login = EncryptedLogin::encrypt(id, &new_login, password)?;
+    let new_login = get_new_encrypted_login(mnemonic, hd_path, id, password)?;
+    stored_wallet.add_encrypted_login(new_login)?;
+
+    write_to_file(filepath, &stored_wallet)
+}
+
+fn store_login_with_multiple_accounts_at_file_overwrite(
+    filepath: &Path,
+    mnemonic: Mnemonic,
+    hd_path: DerivationPath,
+    id: LoginId,
+    password: &UserPassword,
+) -> Result<(), BackendError> {
+    let mut stored_wallet = match load_existing_wallet_at_file(filepath) {
+        Err(BackendError::WalletFileNotFound) => StoredWallet::default(),
+        result => result?,
+    };
 
-    stored_wallet.add_encrypted_login(new_encrypted_login)?;
+    archive_wallet_file()?;
+    let new_login = get_new_encrypted_login(mnemonic, hd_path, id, password)?;
+    stored_wallet.replace_encrypted_login(new_login)?;
```

Comment:
Maybe I'm missing it, but what happens in the case for multiple logins? Will all passwords be updated? Should they?

Also, would be be good with some unit tests for these cases